src/main/java/org/codehaus/mojo/taglist/TagListReport.java

```diff
@@ -435,30 +453,33 @@ private List pruneSourceDirs( List sourceDirectories )
     }
 
     /**
-     * Checks whether the given directory contains Java files.
+     * Checks whether the given directory contains source files.
      * 
      * @param dir the source directory.
-     * @return true if the folder or one of its subfolders contains at least 1 Java file.
+     * @return true if the folder or one of its subfolders contains at least 1 source file that matches includes/excludes.
      */
     private boolean hasSources( File dir )
     {
         boolean found = false;
         if ( dir.exists() && dir.isDirectory() )
         {
-            File[] files = dir.listFiles();
-            for ( int i = 0; i < files.length && !found; i++ )
-            {
-                File currentFile = files[i];
-                if ( currentFile.isFile() && currentFile.getName().endsWith( ".java" ) )
-                {
+            try {
+                if ( ! FileUtils.getFiles( dir, getIncludesCommaSeparated(), getExcludesCommaSeparated() ).isEmpty() ) {
                     found = true;
                 }
-                else if ( currentFile.isDirectory() )
-                {
-                    boolean hasSources = hasSources( currentFile );
-                    if ( hasSources )
-                    {
-                        found = true;
+            } catch (IOException e) {
+                // should never get here
+                getLog().error("Error whilst trying to scan the directory " + dir.getAbsolutePath(), e);
+            }
+            File[] files = dir.listFiles();
+            if ( files != null ) {
+                for ( int i = 0; i < files.length && !found; i++ ) {
```

Comment:
Since this is Java 8 and we do not need the index, you can use enhanced for loops.
However, this is not necessary to change, I will do a Java 8 cleanup in 3.1.0 anyway.

```
for ( File file : files )
{
                    
}
```

Reply:
@bmarwell I fixed this